### PR TITLE
Remove sizeToFit() from SizeCalculable protocol

### PR DESCRIPTION
**Pull request by antoinelamy**
As per the documentation, we should not directly override `sizeToFit()` but rather always only implement `sizeThatFits(_:)` for auto-sizing needs. This PR aim to remove the `sizeToFit()` requirement in the `SizeCalculable` protocol.

>You should not override this method. If you want to change the default sizing information for your view, override the sizeThatFits(_:) instead. That method performs any needed calculations and returns them to this method, which then makes the change.

Internally, UIView has a very basic implementation of the `sizeToFit()` method which is the equivalent of calling `sizeThatFits` with the view bounds so instead of calling it directly, we replicate its behaviour inside PinLayout (the implementation is not gonna change anytime soon anyway).

The original class look like this in assembly pseudo-code:
screen shot 2018-08-13 at 18 53 14

Once converted into a more human readable form, the implementation source code look like this:
```objc
- (void)sizeToFit {
  CGAffineTransform *transform = [self transform];

  if (CGAffineTransformIsIdentity(transform)) {
    CGRect frame = [self frame];
    CGSize size = [self sizeThatFits: CGSizeMake(frame.width, frame.height)];
    [self setFrame: CGRectMake(frame.x, frame.y, size.width, size.height)];
  }
  else {
    CGRect bounds = [self bounds];
    CGSize size = [self sizeThatFits: CGSizeMake(bounds.width, bounds.height)];
    [self setBounds: CGRectMake(bounds.x, bounds.y, size.width, size.height)];
  }
}
```

### CODE REVIEW

File: Sources/Types.swift [hunk 1]
```diff
@@ -164,6 +164,13 @@ public enum FitType {
      sizeThatFits(..) method result.
      */
     case heightFlexible
+
+    /**
+     Adjust the view's size based on it's content size requirements so that it uses the
```
Comment (lucdion) [inline Sources/Types.swift]: Could you also update the Readme.md `sizeToFit()` documentation?

**Comment by lucdion** [comment]
@antoinelamy does this match also the size computed by UILabel.sizeToFit()?

File: Sources/Types.swift (hunk 1 above)
Comment (antoinelamy) [reply Sources/Types.swift]: Done

**Comment by antoinelamy** [comment]
@lucdion sorry it took forever but I finally got the chance to make the requested changes. About your question wether or not it match the size computed by `UILabel.sizeToFit()`, the answer is yes. I also added a test to ensure they both produce the same result with or without a transform applied.

**Comment by lucdion** [comment]
Thanks @antoinelamy 🙂

**Pull request merged**